modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,574 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HexagonMeshTrimmer.h"
+#include "MooseMeshUtils.h"
+
+// C++ includes
+#include <cmath> // provides round, not std::round (see http://www.cplusplus.com/reference/cmath/round/)
+
+registerMooseObject("ReactorApp", HexagonMeshTrimmer);
+
+InputParameters
+HexagonMeshTrimmer::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
+  params.addRequiredParam<MeshGeneratorName>("input", "The input mesh that needs to be trimmed.");
+  params.addRangeCheckedParam<std::vector<unsigned short>>(
+      "trim_peripheral_region",
+      std::vector<unsigned short>(6, 0),
+      "trim_peripheral_region<=1",
+      "Whether the peripheral region on each of the six sides will be trimmed in an assembly "
+      "mesh. See documentation for numbering convention.");
+  params.addParam<BoundaryName>("peripheral_trimming_section_boundary",
+                                "Boundary formed by peripheral trimming.");
+  params.addRangeCheckedParam<short>(
+      "center_trim_starting_index",
+      "center_trim_starting_index>=0 & center_trim_starting_index<12",
+      "Index of the starting center trimming position.");
+  params.addRangeCheckedParam<short>("center_trim_ending_index",
+                                     "center_trim_ending_index>=0 & center_trim_ending_index<12",
+                                     "Index of the ending center trimming position.");
+  params.addParam<BoundaryName>("center_trimming_section_boundary",
+                                "Boundary formed by center trimming.");
+  params.addParam<BoundaryName>("external_boundary", "External boundary of the input mesh.");
```

Comment:
It's not used in testing.

Reply:
This is the external boundary of the input mesh. As we want to move nodes on the external boundary along the boundary instead of in the normal direction.

Reply:
I am adding a test for this parameter.

Also, for center trimming, the new boundary might not be the external boundary

But for the peripheral trimming, the new boundary is always the external boundary.

So I am treating them differently.